File: javascript/rna-transcription/rna-transcription.js
export default class Transcriptor {
  constructor() {
    this.mapper = { G: 'C', C: 'G', T: 'A', A: 'U' };
  }

  toRna(strand) {
    return strand.split('').reduce((rnaStrand, nucleotide) => {
      if (!this.mapper[nucleotide]) {
        throw new Error('Invalid input DNA.');
      }
      return rnaStrand + this.mapper[nucleotide];
    }, '');
  }
}

File: go/rna-transcription/rna_transcription.go
// package strand implements strand exercise from exercism.io
package strand

import "strings"

// ToRNA transcribes a DNA string to a RNA string
func ToRNA(dna string) string {
	mapper := map[string]string{
		"": "",
		"G": "C",
		"C": "G",
		"T": "A",
		"A": "U",
	}

	rna := []string{}

	for _, char := range dna {
		rna = append(rna, mapper[string(char)])
	}

	return strings.Join(rna, "")
}
